src/TcUnit.TestAdapter/Discovery/TestDiscoverer.cs

```diff
@@ -43,12 +43,16 @@ public void DiscoverTests(IEnumerable<string> sources, IDiscoveryContext discove
             try
             {
                 var settings = discoveryContext.RunSettings?.GetTestSettings(TestAdapter.RunSettingsName);
-                var testCaseFilter = new TestCaseFilter(discoveryContext);
+                
+                var testCaseFilter = new TestCaseFilter(discoveryContext, logger);
 
                 var project = TwinCATXAEProject.Load(sources.First());
 
-                var testCases = testRunner.DiscoverTests(project, logger)
-                                            .Where(t => testCaseFilter.MatchTestCase(t));
+                var testCases = testRunner.DiscoverTests(project, logger);
+
+                // currently can't filter test cases unless we have a run context
+                if (discoveryContext is IRunContext)
+                     testCases = testCases.Where(t => testCaseFilter.MatchTestCase(t));
```

Comment:
@philippleidig were you able to use the `/ListTests` option in your environment? I ran into an issue where it was trying to create the filter and the filter was returning no results, so I added the above code.

With this it is possible to call this:
```
> vstest.console.exe /Settings:TcUnit.runsettings  *.tsproj /ListTests
Microsoft (R) Test Execution Command Line Tool Version 16.7.1
Copyright (c) Microsoft Corporation.  All rights reserved.

The following Tests are available:
    PRG_TESTS.fbTestSuite1Instance1.TestCase1B
    PRG_TESTS.fbTestSuite1Instance1.TestCase1C
    PRG_TESTS.fbTestSuite1Instance1.TestCase1D
    PRG_TESTS.fbTestSuite1Instance1.TestCase1E with a space
    PRG_TESTS.fbTestSuite1Instance1.TestCase1F Ignore Case
    PRG_TESTS.fbTestSuite1Instance1.TestCase1A
    PRG_TESTS.fbTestSuite1Instance1.TestCase1G 1*/_-1^&"@!()
    PRG_TESTS.fbTestSuite2Instance1.TestCase2C
    PRG_TESTS.fbTestSuite2Instance1.TestCase2A
    PRG_TESTS.fbTestSuite2Instance1.TestCase2B
    PRG_TESTS.fbTestSuite2Instance2.TestCase2C
    PRG_TESTS.fbTestSuite2Instance2.TestCase2A
    PRG_TESTS.fbTestSuite2Instance2.TestCase2B
```